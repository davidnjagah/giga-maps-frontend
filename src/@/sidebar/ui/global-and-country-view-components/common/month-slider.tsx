import { ChevronLeft, ChevronRight } from '@carbon/icons-react';
import { useStore } from 'effector-react';
import React from 'react';

import { $historyInterval, $isNextHistoryIntervalAvailable, nextHistoryInterval, previousHistoryInterval } from '~/@/sidebar/history-graph.model';
import { formatDateInterval } from '~/lib/date-fns-kit/format-date-interval';


const MonthSlider = () => {
  // Const intervalUnit = useStore($historyIntervalUnit);
  const interval = useStore($historyInterval);
  // Const isCurrentInterval = useStore($isCurrentHistoryInterval);
  const isNextIntervalAvailable = useStore($isNextHistoryIntervalAvailable);
  const formattedInterval = formatDateInterval(
    interval,
    "month",
    false
  );
  return (
    <div className="week_control-container">
      <div className="week_control_style">
        <button
          onClick={() => previousHistoryInterval()}
          type="button"
          className="previous_week_button"
        >
          <ChevronLeft size="32" />
        </button>

        <div className="period-picker__period">{formattedInterval}</div>

        <button
          onClick={() => nextHistoryInterval()}
          disabled={!isNextIntervalAvailable}
          type="button"
          className="period-picker__button"
        >
          <ChevronRight size="32" />
        </button>
      </div>
    </div>
  );
};

export { MonthSlider };
