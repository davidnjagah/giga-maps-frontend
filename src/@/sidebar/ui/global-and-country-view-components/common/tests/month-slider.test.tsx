import { render } from '@testing-library/react';
import { testWrapper } from '~/tests/jest-wrapper';
import { MonthSlider } from '../month-slider';

describe('Month slider', () => {
  it('should render component', () => {
    const { asFragment } = render(testWrapper(<MonthSlider />));
    expect(asFragment).toMatchSnapshot();
  });
});
