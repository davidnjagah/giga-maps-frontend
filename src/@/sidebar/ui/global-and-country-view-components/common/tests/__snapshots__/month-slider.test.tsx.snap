// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Month slider should render component 1`] = `[Function]`;
